contracts/token/ERC721_Enumerable_base.cairo

```diff
@@ -0,0 +1,247 @@
+        syscall_ptr: felt*, 
+        pedersen_ptr: HashBuiltin*, 
+        range_check_ptr
+    }(owner: felt, index: Uint256) -> (token_id: Uint256):
+    alloc_locals
+    # Ensures index argument is less than owner's balance 
+    let (len: Uint256) = ERC721_balanceOf(owner)
+    let (is_lt) = uint256_lt(index, len)
+    assert_not_zero(is_lt)
+
+    let (token_id: Uint256) = owned_tokens.read(owner, index)
+    return (token_id)
+end
+
+#
+# Externals
+#
+
+func ERC721_Enumerable_mint{
+        pedersen_ptr: HashBuiltin*, 
+        syscall_ptr: felt*, 
+        range_check_ptr
+    }(to: felt, token_id: Uint256):
+    _add_token_to_all_tokens_enumeration(token_id)
+    _add_token_to_owner_enumeration(to, token_id)
+    ERC721_mint(to, token_id)
+    return ()
+end
+
+func ERC721_Enumerable_burn{
+        pedersen_ptr: HashBuiltin*, 
+        syscall_ptr: felt*, 
+        range_check_ptr
+    }(token_id: Uint256):
+    let (_from) = ERC721_ownerOf(token_id)
+    _remove_token_from_owner_enumeration(_from, token_id)
+    _remove_token_from_all_tokens_enumeration(token_id)
+    ERC721_burn(token_id)
+    return ()
+end
+
+func ERC721_Enumerable_transferFrom{
+        pedersen_ptr: HashBuiltin*, 
+        syscall_ptr: felt*, 
+        range_check_ptr
+    }(_from: felt, to: felt, token_id: Uint256):
+    _remove_token_from_owner_enumeration(_from, token_id)
+    _add_token_to_owner_enumeration(to, token_id)
+    ERC721_transferFrom(_from, to, token_id)
+    return ()
+end
+
+func ERC721_Enumerable_safeTransferFrom{
+        pedersen_ptr: HashBuiltin*, 
+        syscall_ptr: felt*, 
+        range_check_ptr
+    }(
+        _from: felt, 
+        to: felt, 
+        token_id: Uint256, 
+        data_len: felt,
+        data: felt*
+    ):
+    _remove_token_from_owner_enumeration(_from, token_id)
+    _add_token_to_owner_enumeration(to, token_id)
+    ERC721_safeTransferFrom(_from, to, token_id, data_len, data)
+    return ()
+end
+
+#
+# Internals
+#
+
+func _add_token_to_all_tokens_enumeration{
+        pedersen_ptr: HashBuiltin*, 
+        syscall_ptr: felt*, 
+        range_check_ptr
+    }(token_id: Uint256):
+    alloc_locals
+    let (supply: Uint256) = all_tokens_len.read()
+    # Update all_tokens_list
+    all_tokens_list.write(supply, token_id)
+
+    # Update all_tokens_index
+    all_tokens_index.write(token_id, supply)
+
+    # Update all_tokens_len
+    let (local new_supply: Uint256, _) = uint256_add(supply, Uint256(1, 0))
+    all_tokens_len.write(new_supply)
+    return ()
+end
+
+func _remove_token_from_all_tokens_enumeration{
+        pedersen_ptr: HashBuiltin*, 
+        syscall_ptr: felt*, 
+        range_check_ptr
+    }(token_id: Uint256):
+    alloc_locals
+    let (supply: Uint256) = all_tokens_len.read()
+    let (local index_from_id: Uint256) = all_tokens_index.read(token_id)
+    let (local last_token_id: Uint256) = all_tokens_list.read(supply)
+
+    # Update all_tokens_list i.e. index n => token_id
```

Comment:
I think these comments are simply describing the line below, which is not necessary. I think it's worth copying or adapting the comments from the original implementation which clarifies the reasoning behind this operations.